clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/ClusteredStore.java

```diff
@@ -152,44 +155,66 @@ public boolean containsKey(final K key) throws StoreAccessException {
   @Override
   public PutStatus put(final K key, final V value) throws StoreAccessException {
     putObserver.begin();
-    PutOperation<K, V> operation = new PutOperation<K, V>(key, value, timeSource.getTimeMillis());
-    ByteBuffer payload = codec.encode(operation);
-    Chain chain = storeProxy.getAndAppend(key.hashCode(), payload);
-    ResolvedChain<K, V> resolvedChain = resolver.resolve(chain, key, timeSource.getTimeMillis());
-    if(resolvedChain.getResolvedResult(key) == null) {
-      putObserver.end(StoreOperationOutcomes.PutOutcome.PUT);
-      return PutStatus.PUT;
-    } else {
-      putObserver.end(StoreOperationOutcomes.PutOutcome.REPLACED);
-      return PutStatus.UPDATE;
+    PutStatus status = silentPut(key, value);
+    switch (status) {
+      case PUT:
+        putObserver.end(StoreOperationOutcomes.PutOutcome.PUT);
+        break;
+      case UPDATE:
+        putObserver.end(StoreOperationOutcomes.PutOutcome.REPLACED);
+        break;
+      case NOOP:
+        putObserver.end(StoreOperationOutcomes.PutOutcome.NOOP);
+        break;
+      default:
+        throw new IllegalArgumentException("Invalid put status: " + status);
+    }
+    return status;
+  }
+
+  private PutStatus silentPut(final K key, final V value) throws StoreAccessException {
+    try {
+      PutOperation<K, V> operation = new PutOperation<K, V>(key, value, timeSource.getTimeMillis());
+      ByteBuffer payload = codec.encode(operation);
+      Chain chain = storeProxy.getAndAppend(key.hashCode(), payload);
+      ResolvedChain<K, V> resolvedChain = resolver.resolve(chain, key, timeSource.getTimeMillis());
+      if(resolvedChain.getResolvedResult(key) == null) {
+        return PutStatus.PUT;
+      } else {
+        return PutStatus.UPDATE;
+      }
+    } catch (RuntimeException re) {
+      handleRuntimeException(re);
+      return null;
```

Comment:
As bizzare as it may sound, I would return `PutStatus.NOOP` here. This will make zero difference at runtime but would help code checkers understand this method will never actually return null.